src/emqx_protocol.erl

```diff
@@ -84,28 +85,29 @@
 -spec(init(map(), list()) -> state()).
 init(#{peername := Peername, peercert := Peercert, sendfun := SendFun}, Options) ->
     Zone = proplists:get_value(zone, Options),
-    #pstate{zone          =  Zone,
-            sendfun       =  SendFun,
-            peername      =  Peername,
-            peercert      =  Peercert,
-            proto_ver     =  ?MQTT_PROTO_V4,
-            proto_name    =  <<"MQTT">>,
-            client_id     =  <<>>,
-            is_assigned   =  false,
-            conn_pid      =  self(),
-            username      =  init_username(Peercert, Options),
-            is_super      =  false,
-            clean_start   =  false,
-            topic_aliases =  #{},
-            packet_size   =  emqx_zone:get_env(Zone, max_packet_size),
-            mountpoint    =  emqx_zone:get_env(Zone, mountpoint),
-            is_bridge     =  false,
-            enable_ban    =  emqx_zone:get_env(Zone, enable_ban, false),
-            enable_acl    =  emqx_zone:get_env(Zone, enable_acl),
-            recv_stats    =  #{msg => 0, pkt => 0},
-            send_stats    =  #{msg => 0, pkt => 0},
-            connected     =  false,
-            ignore_loop   =  emqx_config:get_env(mqtt_ignore_loop_deliver, false)}.
+    #pstate{zone            = Zone,
+            sendfun         = SendFun,
+            peername        = Peername,
+            peercert        = Peercert,
+            proto_ver       = ?MQTT_PROTO_V4,
+            proto_name      = <<"MQTT">>,
+            client_id       = <<>>,
+            is_assigned     = false,
+            conn_pid        = self(),
+            username        = init_username(Peercert, Options),
+            is_super        = false,
+            clean_start     = false,
+            topic_aliases   = #{},
+            packet_size     = emqx_zone:get_env(Zone, max_packet_size),
+            mountpoint      = emqx_zone:get_env(Zone, mountpoint),
+            is_bridge       = false,
+            enable_ban      = emqx_zone:get_env(Zone, enable_ban, false),
+            enable_acl      = emqx_zone:get_env(Zone, enable_acl),
+            acl_deny_action = emqx_zone:get_env(Zone, acl_deny_action, ignore),
```

Comment:
Load 'acl_deny_action ' configuration for the first time, Reduce duplicate read configuration， I learned😁😁